src/cli/core/GeneratedFileWriter.ts

```diff
@@ -38,7 +39,7 @@ export class GeneratedFileWriter {
   public write = async (): Promise<void> => {
     // Check CLI inputs
     try {
-      this.validateCliOptions();
+      await this.validateCliOptions();
```

Comment:
I added an `await` here so that thrown errors don't result in process termination before the logger has had a chance to print to the screen.